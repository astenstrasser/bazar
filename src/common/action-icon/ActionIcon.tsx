import React from "react";

function ActionIcon(props: {
  onClick: () => void;
  src: any;
  alt: string;
}): React.ReactElement {
  return (
    <a onClick={props.onClick}>
      <img className="action-icon" src={props.src} alt={props.alt} />
    </a>
  );
}

export default ActionIcon;
