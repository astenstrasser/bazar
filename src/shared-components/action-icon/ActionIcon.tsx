import React from "react";
import { Icon } from "../types";
import "./ActionIcon.css";
import LeftArrow from "../../assets/left-arrow.svg";
import PlusIcon from "../../assets/plus.svg";

function ActionIcon(props: {
  onClick: () => void;
  type: Icon;
  alt: string;
}): React.ReactElement {
  function getIcon() {
    switch (props.type) {
      case Icon.LeftArrow:
        return LeftArrow;
      case Icon.Plus:
        return PlusIcon;
    }
  }
  return (
    <a onClick={props.onClick}>
      <img className="action-icon" src={getIcon()} alt={props.alt} />
    </a>
  );
}

export default ActionIcon;
