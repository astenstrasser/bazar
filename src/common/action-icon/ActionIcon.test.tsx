import { render } from "@testing-library/react";
import React from "react";
import ActionIcon, { Icon } from "./ActionIcon";

test("renders snapshot", () => {
  const { container } = render(
    <ActionIcon
      onClick={() => {
        console.log("test");
      }}
      alt={"alt"}
      type={Icon.LeftArrow}
    />
  );
  expect(container).toMatchSnapshot();
});
