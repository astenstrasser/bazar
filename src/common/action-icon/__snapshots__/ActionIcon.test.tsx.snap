// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div>
  <a>
    <img
      alt="alt"
      class="action-icon"
      src="left-arrow.svg"
    />
  </a>
</div>
`;
